pkg/apis/core/v1alpha1/podlogstream_types.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2020 The Tilt Dev Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+
+	"github.com/tilt-dev/tilt-apiserver/pkg/server/builder/resource"
+	"github.com/tilt-dev/tilt-apiserver/pkg/server/builder/resource/resourcestrategy"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PodLogStream
+//
+// Streams logs from a pod on Kubernetes into the core Tilt engine.
+//
+// +k8s:openapi-gen=true
+type PodLogStream struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PodLogStreamSpec   `json:"spec,omitempty"`
+	Status PodLogStreamStatus `json:"status,omitempty"`
+}
+
+// PodLogStreamList
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+type PodLogStreamList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+
+	Items []PodLogStream `json:"items"`
+}
+
+// PodLogStreamSpec defines the desired state of PodLogStream
+//
+// Translated into a PodLog query to the current Kubernetes cluster:
+// https://pkg.go.dev/k8s.io/api/core/v1#PodLogOptions
+//
+// TODO(nick): Should all Kubernetes types have an object that describes
```

Comment:
I wonder if we should have a helper type that's a tuple of `Kind, Namespace, Name` - kind of a `types.NamespacedName++`

Reply:
oh i think this is weirder tho - this is like a KUBECONFIG, context, and namespace